inf-terraform-azure/files/Makefile

```diff
@@ -0,0 +1,242 @@
+	@TF_IN_AUTOMATION=1 TF_WORKSPACE="$(TF_WORKSPACE)" terraform show -no-color -json | tee "$(INSTALL_REPORT_HOME)/tf_show.log" 1>/dev/null
+
+.PHONY: smoke-test
+# Run (post-deployment) tests.
+smoke-test: export ARM_SUBSCRIPTION_ID := $(AZURE_SUBSCRIPTION_ID)
+smoke-test: export ARM_TENANT_ID       := $(AZURE_TENANT_ID)
+smoke-test: export ARM_CLIENT_ID       := $(AZURE_CLIENT_ID)
+smoke-test: export ARM_CLIENT_SECRET   := $(AZURE_CLIENT_SECRET)
+smoke-test: install-test-deps
+	$(call check_defined,AZURE_SUBSCRIPTION_ID)
+	$(call check_defined,AZURE_TENANT_ID)
+	$(call check_defined,AZURE_CLIENT_ID)
+	$(call check_defined,AZURE_CLIENT_SECRET)
+
+	sh ./lib/scripts/createstackoutputs2yml.sh
+	inspec_profiles=$$(ls -1 ./test/integration); \
+	for profile in $$inspec_profiles; do \
+		JSON_VARS_FILE="$(PWD)/terraform.tfvars.json" bundle exec cinc-auditor exec ./test/integration/$$profile --reporter=cli junit2:build/test-results/test/$$profile.xml json:reports/install/data/inspec/post-install/$$profile.json --target azure://; \
+	done
+
+.PHONY: install-report
+install-report:
+	awk '/Creation complete/ && !/terraform-data/ {print}' "$(INSTALL_REPORT_HOME)/tf_apply.log" > $(INSTALL_REPORT_HOME)/tf_created.log
+
+.PHONY: describe
+# Describe infrastructure.
+describe: init-terraform
+	TF_IN_AUTOMATION=1 TF_WORKSPACE="$(TF_WORKSPACE)" terraform output -json | tee outputs.json
+
+.PHONY: destroy
+# Destroy infrastructure.
+destroy: init-terraform
+destroy: export ARM_SUBSCRIPTION_ID := $(AZURE_SUBSCRIPTION_ID)
+destroy: export ARM_TENANT_ID       := $(AZURE_TENANT_ID)
+destroy: export ARM_CLIENT_ID       := $(AZURE_CLIENT_ID)
+destroy: export ARM_CLIENT_SECRET   := $(AZURE_CLIENT_SECRET)
+destroy: init-terraform
+	$(call check_defined,AZURE_SUBSCRIPTION_ID)
+	$(call check_defined,AZURE_TENANT_ID)
+	$(call check_defined,AZURE_CLIENT_ID)
+	$(call check_defined,AZURE_CLIENT_SECRET)
+
+	TF_IN_AUTOMATION=1 TF_WORKSPACE="$(TF_WORKSPACE)" terraform destroy -auto-approve
+
+.PHONY: install-dev-deps
+# Install development dependencies.
+install-dev-deps: install-git-pre-commit-hooks
+
+.PHONY: install-git-pre-commit-hooks
+# Install Git pre-commit hooks.
+install-git-pre-commit-hooks:
+	pre-commit install --overwrite
+
+.PHONY: install-ruby-gems
+# Install Ruby gems specified in Gemfile.
+install-ruby-gems:
+	BUNDLE_SILENCE_ROOT_WARNING=true bundle config --local path $(GEMS_HOME)
+	# see https://github.com/rubygems/rubygems/issues/4466 to get rid of error messages in Jenkins
+	BUNDLE_SILENCE_ROOT_WARNING=true TMPDIR=./vendor/tmp bundle install --jobs=8
+
+.PHONY: install-python-env
+# Install python virtual environment based on Pipfile
+install-python-env:
+	CI=true PIPENV_VENV_IN_PROJECT=true pipenv install
+
+.PHONY: init-terraform
+# Install Terraform workspace.
+init-terraform:
+	echo 1 | terraform init -backend=true -force-copy -input=false -backend-config="backends/$(TF_BACKEND).tfbackend"
```

Comment:
@matthiasbremser 
Please move the *.tfbackend files into folder environments and name those
[env].tfbackend.config